File: fixtures/scale/typescript/large/src/ciphers/Augustus.ts
// Additive byte-shift cipher (augustus).
import { Bytes } from '../util/Bytes';

export class Augustus {
  static encrypt(text: string, key: number): string {
    const shift = (key + 5) % 256;
    return Bytes.fromCodes(Bytes.toCodes(text).map((v) => v + shift));
  }

  static decrypt(text: string, key: number): string {
    const shift = (key + 5) % 256;
    return Bytes.fromCodes(Bytes.toCodes(text).map((v) => v - shift));
  }
}

File: fixtures/scale/lua/large/ciphers/augustus.lua
local bytes = require("util.bytes")

local function augustus_encrypt(text, key)
  local shift = (key + 5) % 256
  local codes = bytes.to_codes(text)
  for i, c in ipairs(codes) do codes[i] = c + shift end
  return bytes.from_codes(codes)
end

local function augustus_decrypt(text, key)
  local shift = (key + 5) % 256
  local codes = bytes.to_codes(text)
  for i, c in ipairs(codes) do codes[i] = c - shift end
  return bytes.from_codes(codes)
end

return { augustus_encrypt = augustus_encrypt, augustus_decrypt = augustus_decrypt }
